src/TestFramework/TestFramework/Attributes/TestMethod/TimeoutAttribute.cs

```diff
@@ -52,5 +59,10 @@ public bool CooperativeCancellation
         }
     }
 
+    /// <summary>
+    /// Gets a value indicating whether the instance has the correct test timeout signature.
+    /// </summary>
+    internal bool HasCorrectTimeout { get; private set; }
```

Comment:
My personal preference here is to keep the constructors as they were before and switch this to:

```suggestion
    internal bool HasCorrectTimeout => Timeout > 0;
```